database/rrdhost.c

```diff
@@ -634,11 +635,11 @@ int rrd_init(char *hostname, struct rrdhost_system_info *system_info) {
     if (gap_when_lost_iterations_above < 1)
         gap_when_lost_iterations_above = 1;
 
-#ifdef ENABLE_DBENGINE
     if (unlikely(sql_init_database())) {
-        return 1;
+        if (default_rrd_memory_mode == RRD_MEMORY_MODE_DBENGINE)
+            return 1;
+        info("Failed to initialize SQLITE metadata; continuing operation since memory mode is not dbengine");
```

Comment:
Sure I will rephrase it. It is not exactly skipping since that would imply it didn't try at all because memory mode is db engine, but it can be improved.